teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -145,55 +285,79 @@ interface Branch {
     val isDefault: Boolean
 }
 
+interface BuildCanceledInfo {
+    val user: User?
+    val cancelDate: Date
+}
+
 interface Build {
     val id: BuildId
-    val buildTypeId: BuildConfigurationId
-    val buildNumber: String
-    val status: BuildStatus
+    val buildConfigurationId: BuildConfigurationId
+    val buildNumber: String?
+    val status: BuildStatus?
```

Comment:
`status` and `state` look similar to me. Maybe we can merge them into a single enum?

Reply:
Nope. `state` tell us current point of a build lifecycle . `status` tell us a build result.
Also at some point of time we'll need to support failed to start builds

Reply:
But `status` makes sense only if `state` is `FINISHED`, isn't it? Or `status` may be `ERROR` while a build is still running?

Reply:
yes, the status is present after a build was started